python/ray/serve/controller.py

```diff
@@ -315,6 +320,12 @@ def _put_serve_snapshot(self) -> None:
             val[deployment_name] = entry
         self.snapshot_store.put(SNAPSHOT_KEY, json.dumps(val).encode("utf-8"))
 
+    def _recover_config_from_checkpoint(self):
+        checkpoint = self.kv_store.get(CONFIG_CHECKPOINT_KEY)
+        if checkpoint is not None:
+            self.deployment_timestamp, config = cloudpickle.loads(checkpoint)
```

Comment:
any reason we can't use just vanilla pickle here? pickle is least powerful

Reply:
We probably can. I was pattern-matching the checkpoint from [`DeploymentStateManager`](https://github.com/ray-project/ray/blob/e433424796d35eefada0cec34ede52b07d14e25c/python/ray/serve/deployment_state.py#L1661-L1671). Should I change this to `pickle`?

Reply:
If you can use it, it's preferred so we don't accidentally serialize code as well